crates/uv-interpreter/src/virtualenv.rs

```diff
@@ -0,0 +1,156 @@
+use std::{
+    env, io,
+    path::{Path, PathBuf},
+};
+
+use fs_err as fs;
+use pypi_types::Scheme;
+use thiserror::Error;
+use tracing::{debug, info};
+
+/// The layout of a virtual environment.
+#[derive(Debug)]
+pub struct VirtualEnvironment {
```

Comment:
We might want to make this struct more powerful in the future. Opting not to make more changes now though.